src/primitive_file_win.cc

```diff
@@ -405,111 +403,104 @@ PRIMITIVE(stat) {
 }
 
 PRIMITIVE(unlink) {
-  ARGS(cstring, pathname);
-  char path[MAX_PATH];
-  auto error = get_absolute_path(process, pathname, path);
-  if (error) return error;
+  ARGS(StringOrSlice, pathname);
+  BLOB_TO_ABSOLUTE_PATH(path, pathname);
 
-  int result = unlink(path);
+  int result = _wunlink(path);
   if (result < 0) return return_open_error(process, errno);
   return process->program()->null_object();
 }
 
 PRIMITIVE(rmdir) {
-  ARGS(cstring, pathname);
-  char path[MAX_PATH];
-  auto error = get_absolute_path(process, pathname, path);
-  if (error) return error;
+  ARGS(StringOrSlice, pathname);
+  BLOB_TO_ABSOLUTE_PATH(path, pathname);
 
-  if (RemoveDirectory(path) == 0) WINDOWS_ERROR;
+  if (RemoveDirectoryW(path) == 0) WINDOWS_ERROR;
   return process->program()->null_object();
 }
 
 PRIMITIVE(rename) {
-  ARGS(cstring, old_name, cstring, new_name);
-  int result = rename(old_name, new_name);
+  ARGS(StringOrSlice, old_name_blob, StringOrSlice, new_name_blob);
+  WideCharAllocationManager allocation1(process), allocation2(process);
+  wchar_t* old_name = allocation1.to_wcs(&old_name_blob);
+  wchar_t* new_name = allocation2.to_wcs(&new_name_blob);
+  // TODO(erik): Should this use absolute path?
+  int result = _wrename(old_name, new_name);
   if (result < 0) return return_open_error(process, errno);
   return process->program()->null_object();
 }
 
 PRIMITIVE(chdir) {
-  ARGS(cstring, pathname);
-  size_t pathname_length = strlen(pathname);
-
-  if (pathname_length == 0) INVALID_ARGUMENT;
+  ARGS(StringOrSlice, pathname);
+  if (pathname.length() == 0) INVALID_ARGUMENT;
+  BLOB_TO_ABSOLUTE_PATH(path, pathname);
 
-  char path[MAX_PATH];
-  auto error = get_absolute_path(process, pathname, path);
-  if (error) return error;
-
-  char* copy = strdup(path);
-  if (!copy) MALLOC_FAILED;
+  wchar_t* copy = wcsdup(path);
 
   process->set_current_directory(copy);
 
   return process->program()->null_object();
 }
 
 PRIMITIVE(mkdir) {
-  ARGS(cstring, pathname, int, mode);
-  char path[MAX_PATH];
-  auto error = get_absolute_path(process, pathname, path);
-  if (error) return error;
+  ARGS(StringOrSlice, pathname, int, mode);
+  BLOB_TO_ABSOLUTE_PATH(path, pathname);
 
-  int result = CreateDirectory(path, NULL);
+  int result = CreateDirectoryW(path, NULL);
   if (result == 0) WINDOWS_ERROR;
   return process->program()->null_object();
 }
 
 PRIMITIVE(mkdtemp) {
-  ARGS(cstring, prefix);
+  ARGS(StringOrSlice, prefix_blob);
   DWORD ret;
 
+  WideCharAllocationManager allocation(process);
+  wchar_t* prefix = allocation.to_wcs(&prefix_blob);
```

Comment:
We can't give the system relative paths.
Once we know that the path isn't `/tmp/` we need to make it absolute.